atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt

```diff
@@ -784,6 +785,18 @@ class AtomicFUTransformer(
         }
 
         private fun fixupLoadedAtomicVar(f: FieldInfo, ld: FieldInsnNode): AbstractInsnNode? {
+            if (f.fieldType == REF_TYPE && ld.next.opcode == DUP && ld.next.next.opcode == IFNONNULL) {
```

Comment:
This check seems to be very restrictive and the code does not work with array values. Consider the following test:

```
    private val a = atomicArrayOfNulls<Any?>(10)

    @Test
    fun testArrayValueUncheckedCast() {
        a[0].value = "OK"
        assertEquals("OK", (a[0] as AtomicRef<String>).value)
    }
```

It produces Ok diagnostics ("Unsupported operation on atomic variable"). However, the following test:

```
    private val a = atomicArrayOfNulls<Any?>(10)

    @Test
    fun testArrayValueUncheckedCast() {
        a[0].value = "OK"
        assertEquals("OK", a[0].getString())
    }

    inline fun <T> AtomicRef<T>.getString() =
        (this as AtomicRef<String>).value
```

Completely crashes transformer: 

```
[main] ERROR kotlinx.atomicfu.transformer.AtomicFUTransformer - UncheckedCastTest.kt: kotlinx.atomicfu.test.UncheckedCastTest::testArrayValueUncheckedCast: Failed to transform: java.lang.ClassCastException: org.objectweb.asm.tree.VarInsnNode cannot be cast to org.objectweb.asm.tree.MethodInsnNode
java.lang.ClassCastException: org.objectweb.asm.tree.VarInsnNode cannot be cast to org.objectweb.asm.tree.MethodInsnNode
        at kotlinx.atomicfu.transformer.AtomicFUTransformer$TransformerMV.fixupInvokeVirtual(AtomicFUTransformer.kt:671)
        at kotlinx.atomicfu.transformer.AtomicFUTransformer$TransformerMV.fixupLoadedAtomicVar(AtomicFUTransformer.kt:805)
...
```

It seems a little bit of extra effort to generalize this code a bit so that casting of array elements is supported, too.